src/api/java/mezz/jei/api/IModPlugin.java

```diff
@@ -45,11 +50,25 @@ default void register(IModRegistry registry) {
 	}
 
 	/**
-	 * Called when jei's runtime features are available, after all mods have registered.
+	 * Called when jei's runtime features are available, after all mods have
+	 * registered.
 	 *
 	 * @since JEI 2.23.0
 	 */
 	default void onRuntimeAvailable(IJeiRuntime jeiRuntime) {
 
 	}
+
+	/**
+	 * Register additional sorting option.
+	 *
+	 * @since JEI 4.?.0
+	 */
+	default void registerSorter(String name, Comparator<ItemStack> comparator) {
```

Comment:
This class gets implemented by mods, so this method is in the wrong place.
This method should go in `IModRegsitry`, which is implemented by JEI and passed to mod plugins.